shared/storage.ts

```diff
@@ -20,17 +21,42 @@ const vcKeyRegExp = new RegExp(VC_ITEM_STORE_KEY_REGEX);
 const vcDirectoryPath = `${DocumentDirectoryPath}/inji/VC`;
 
 class Storage {
-  static getItem = async (key: string) => {
-    if (vcKeyRegExp.exec(key)) {
-      const path = getFilePath(key);
-      return await readFile(path, 'utf8');
+  static getItem = async (key: string, encryptionKey: string) => {
+    try {
+      if (vcKeyRegExp.exec(key)) {
+        const path = getFilePath(key);
+        const data = await readFile(path, 'utf8');
+
+        const encryptedHMACofCurrentVC = await MMKV.getItem(key);
+        const HMACofCurrentVC = CryptoJS.AES.decrypt(
+          encryptedHMACofCurrentVC,
+          encryptionKey
+        ).toString(CryptoJS.enc.Utf8);
+
+        const HMACofVC = CryptoJS.HmacSHA256(encryptionKey, data).toString();
+        if (HMACofVC === HMACofCurrentVC) {
```

Comment:
can we refactor this as 
`return (HMACofVC === HMACofCurrentVC) ? data : null`

Reply:
Refactored.